sdks/python/apache_beam/io/vcfio_test.py

```diff
@@ -0,0 +1,513 @@
+
+    self.assertEqual(base_variant, equal_variant)
+    self.assertNotEqual(base_variant, different_calls)
+    self.assertNotEqual(base_variant, missing_field)
+
+  @unittest.skipIf(VCF_FILE_DIR_MISSING, 'VCF test file directory is missing')
+  def test_read_single_file_large(self):
+    test_data_conifgs = [
+        {'file': 'valid-4.0.vcf', 'num_records': 5},
+        {'file': 'valid-4.0.vcf.gz', 'num_records': 5},
+        {'file': 'valid-4.0.vcf.bz2', 'num_records': 5},
+        {'file': 'valid-4.1-large.vcf', 'num_records': 9882},
+        {'file': 'valid-4.2.vcf', 'num_records': 13},
+    ]
+    for config in test_data_conifgs:
+      read_data = self._read_records(
+          get_full_file_path(config['file']))
+      self.assertEqual(config['num_records'], len(read_data))
+
+  @unittest.skipIf(VCF_FILE_DIR_MISSING, 'VCF test file directory is missing')
+  def test_read_file_pattern_large(self):
+    read_data = self._read_records(
+        os.path.join(get_full_dir(), 'valid-*.vcf'))
+    self.assertEqual(9900, len(read_data))
+    read_data_gz = self._read_records(
+        os.path.join(get_full_dir(), 'valid-*.vcf.gz'))
+    self.assertEqual(9900, len(read_data_gz))
+
+  def test_single_file_no_records(self):
+    self.assertEqual(
+        [], self._create_temp_file_and_read_records(['']))
+    self.assertEqual(
+        [], self._create_temp_file_and_read_records(['\n', '\r\n', '\n']))
+    self.assertEqual(
+        [], self._create_temp_file_and_read_records(_SAMPLE_HEADER_LINES))
+
+  def test_single_file_verify_details(self):
+    variant_1, vcf_line_1 = self._get_sample_variant_1()
+    read_data = self._create_temp_file_and_read_records(
+        _SAMPLE_HEADER_LINES + [vcf_line_1])
+    self.assertEqual(1, len(read_data))
+    self.assertEqual(variant_1, read_data[0])
+    variant_2, vcf_line_2 = self._get_sample_variant_2()
+    variant_3, vcf_line_3 = self._get_sample_variant_3()
+    read_data = self._create_temp_file_and_read_records(
+        _SAMPLE_HEADER_LINES + [vcf_line_1, vcf_line_2, vcf_line_3])
+    self.assertEqual(3, len(read_data))
+    self._assert_variants_equal([variant_1, variant_2, variant_3], read_data)
+
+  def test_file_pattern_verify_details(self):
+    variant_1, vcf_line_1 = self._get_sample_variant_1()
+    variant_2, vcf_line_2 = self._get_sample_variant_2()
+    variant_3, vcf_line_3 = self._get_sample_variant_3()
+    tmpdir = self._new_tempdir()
+    self._create_temp_vcf_file(_SAMPLE_HEADER_LINES + [vcf_line_1],
+                               tmpdir=tmpdir)
+    self._create_temp_vcf_file(_SAMPLE_HEADER_LINES + [vcf_line_2, vcf_line_3],
+                               tmpdir=tmpdir)
+    read_data = self._read_records(os.path.join(tmpdir, '*.vcf'))
+    self.assertEqual(3, len(read_data))
+    self._assert_variants_equal([variant_1, variant_2, variant_3], read_data)
+
+  @unittest.skipIf(VCF_FILE_DIR_MISSING, 'VCF test file directory is missing')
```

Comment:
Why do we need this skipping feature - in what case will it be missing, and can this lead to a situation where for whatever reason it goes missing and nobody notices and test coverage is silently reduced?

Reply:
Since the VCF files are very large, we chose not to include them in the distribution. I created test cases with smaller temp files to test the same behavior for the distribution.